packages/amplify_auth_plugin_interface/lib/src/SignOut/SignOutOptions.dart

```diff
@@ -13,10 +13,21 @@
  * permissions and limitations under the License.
  */
 
-abstract class SignOutOptions {
-  const SignOutOptions();
+/// The shared sign out options among all Auth plugins
+class SignOutOptions {
```

Comment:
Should we make a CognitoSignOutOptions class which extends this at the plugin (as opposed to the category) level?

Reply:
I wasn't sure if this would apply to all plugins either so I followed Amplify-Android and Amplify-iOS.

[Android has it on the non-cognito-specific class](https://github.com/aws-amplify/amplify-android/blob/main/core/src/main/java/com/amplifyframework/auth/options/AuthSignOutOptions.java#L25). iOS expects a [SignOutOptions from the SDK](https://github.com/aws-amplify/aws-sdk-ios/blob/2ce94a7fda175ed6ffd86c993b0ace219b926ac2/AWSAuthSDK/Sources/AWSMobileClient/AWSMobileOptions.swift#L52), which has globalSignOut on it.